proto-lens-protobuf-types/src/Data/ProtoLens/Descriptor.hs

```diff
@@ -0,0 +1,17 @@
+-- | Functions for interacting with message descriptors.
+module Data.ProtoLens.Descriptor
+    ( DescriptorProto
+    , messageDescriptor
+    ) where
+
+import Data.ProtoLens
+import Data.Proxy (Proxy)
+import Proto.Google.Protobuf.Descriptor
+
+-- | The protocol buffer message descriptor for this type.
+messageDescriptor :: Message a => Proxy a -> DescriptorProto
```

Comment:
Have you not warmed up to TypeApplications? It seems like a syntactically lighter option? Or does it somehow not work here?

Reply:
Good idea; done.

Maybe we should do that for the methods in `Data.ProtoLens.Message` too (`messageName` and `packedMessageDescriptor`)?  I'm not sure, so I left them alone for now.